### ISSUE TITLE
Deterministic Generators Do Not Compose Well With Gen.oneOf

### ISSUE THREAD

**Issue by adamgfraser** [zio/zio#5616]
```scala
  val sometimesNone: Gen[Random, Option[String]] = Gen.option(Gen.fromIterable("true", "false"))

  override def spec =
    suite("sometimesNone")(
      testM("should sometimes fail") {
        checkAll(sometimesNone)(maybe => assert(maybe)(isSome))
      }
    )
```

This never generates a `None` value.

**Comment by scottweaver** [comment]
I have currently come upon a similar issue where what should be a random value gets "stuck" and produces the same value over and over again.  In fact, I have cases where the specific value gets stuck even between multiple runs of the test case.

Not sure if this is related to this, but it feels similar.

**Comment by adamgfraser** [comment]
@scottweaver Are you using deterministic generators? Do you have a small reproducer?

**Comment by scottweaver** [comment]
Yeah, I am using `oneOf`, `fromIterable`.  As for a small repoducer, I don't have one immediately.  

I noticed this behavior in a framework I built that creates Gens that emit JSON based on an OpenAPI schema.  I would need to see if I can break it into something I can post.  

What is really odd is that I can have to fields being generated and one randomizes fine and the other gets stuck.

**Comment by adamgfraser** [comment]
Okay. I am working on something for this.
